ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -340,6 +346,10 @@ private void pollOnMainThread(long timeoutMs, List<NetworkSend> sends) throws IO
       metrics.selectorReadyKeyCount.inc(keys.size());
       Iterator<SelectionKey> iter = keys.iterator();
       while (iter.hasNext()) {
+        if (networkConfig.selectorMaxRunningTimeInMs != -1
```

Comment:
It seems like one of these configs applies only for main thread polling and one applies for executor pool polling